spring-integration-core/src/main/java/org/springframework/integration/channel/NullChannel.java

```diff
@@ -27,32 +32,144 @@
  * All receive() calls will return <em>null</em>, and all send() calls
  * will return <em>true</em> although no action is performed.
  * Note however that the invocations are logged at debug-level.
- * 
+ *
  * @author Mark Fisher
+ * @author Gary Russell
  */
-public class NullChannel implements PollableChannel {
+@ManagedResource
+public class NullChannel implements PollableChannel, MessageChannelMetrics, NamedComponent {
 
 	private final Log logger = LogFactory.getLog(this.getClass());
 
+	private final ChannelSendMetrics metrics = new ChannelSendMetrics("nullChannel");
```

Comment:
Some smell is here.
I always can create my own bean:
````java
@Bean
public PollableChannel myNullChannel() {
  return new NullChannel();
}
````
And I will be surprised when the statst and metrics for `nullChannel` MBean will be collected (or overrided ?) from everywhere not only from that endpoint which is supplied with this my `NullChannel` isntance.
Seems for me `BeanNameAware` should be there as well.